### DISCUSSION TITLE
issue: cant reset input file

### DISCUSSION THREAD

**Discussion by rendivs925** [react-hook-form/react-hook-form#12183] [Q&A]
### Version Number

latest and greatest

### Codesandbox/Expo snack

none

### Steps to reproduce

![screenshot-20240727-232426Z-all](https://github.com/user-attachments/assets/ea934955-faad-4411-968b-6c10b182183d)

### Expected behaviour

the input file should be no choosen file after reset

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rendivs925** [comment]
![screenshot-20240727-232705Z-all](https://github.com/user-attachments/assets/7ca94411-d3d5-49cd-89a7-334f08ac7412)

**Comment by alejandro-almeida-improving** [comment]
I have the same issue, any workaround?

**Comment by samuelho-dev** [comment]
having this same issue, as well as array fields

**Comment by raky291** [comment]
The workaround was to create a controlled file input component, the library cannot reset the input if it's not controlled.

https://medium.com/trabe/controlled-file-input-components-in-react-3f0d42f901b8
